airo-robots/airo_robots/manipulators/hardware/ur_rtde.py

```diff
@@ -1,20 +1,144 @@
 import time
+from multiprocessing import Array, Process, Value
 from typing import Optional
 
 import numpy as np
 from airo_robots.awaitable_action import AwaitableAction
 from airo_robots.grippers import ParallelPositionGripper
 from airo_robots.manipulators.position_manipulator import ManipulatorSpecs, PositionManipulator
 from airo_spatial_algebra import SE3Container
-from airo_typing import HomogeneousMatrixType, JointConfigurationType
-from loguru import logger
+from airo_typing import HomogeneousMatrixType, JointConfigurationType, WrenchType
 from rtde_control import RTDEControlInterface
 from rtde_receive import RTDEReceiveInterface
 
 RotVecPoseType = np.ndarray
 """ a 6D pose [tx,ty,tz,rotvecx,rotvecy,rotvecz]"""
 
 
+def _torque_worker(
+    ur_ip,
+    target_pos_shared,  # Array('d', 6)
+    pos_cache_shared,  # Array('d', 6)
+    tcp_cache_shared,  # Array('d', 6)
+    gripper_cache_shared,  # Value('d')
+    running_flag,  # Value('b')
+    use_gripper=False,
+    log_path: str | None = "torque_log.csv",
+):
+
+    rtde_c = RTDEControlInterface(ur_ip, 500.0)
+    rtde_r = RTDEReceiveInterface(ur_ip, 500.0)
+
+    gripper = None
+    if use_gripper:
+        from airo_robots.grippers import Robotiq2F85
+
+        gripper = Robotiq2F85(ur_ip)
+
+    # UR3E_MAX_TORQUE = np.array([72.0, 72.0, 72.0, 26.0, 18.0, 18.0])/6
+    # max_torque = UR3E_MAX_TORQUE
+    # Kp = max_torque / np.array([0.1, 0.1, 0.1, 0.1, 0.07, 0.125])
+    # Kp *= 2.5
+    # Kd = max_torque / (np.pi * 0.25)
+    # Kd *= 2
+
+    # UR3E_MAX_TORQUE = np.array([56., 56., 28., 9., 9., 9.]) * 0.5  # 或 0.7
+    # max_torque = UR3E_MAX_TORQUE
+    #
+    # Kp = UR3E_MAX_TORQUE / np.array([0.35, 0.35, 0.35, 0.6, 0.6, 1.0])
+    # Kd = UR3E_MAX_TORQUE / (np.pi * 0.5)
+    # need tuning
+    UR3E_MAX_TORQUE = np.array([56.0, 56.0, 28.0, 9.0, 9.0, 9.0])
```

Comment:
It would be nice to have this as a constant, e.g., in `URrtde`, similar to the `ManipulatorSpecs`.